embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceDeliveryRetryManager.kt

```diff
@@ -42,16 +47,23 @@ internal class EmbraceDeliveryRetryManager(
      */
     override fun scheduleForRetry(request: ApiRequest, payload: ByteArray) {
         logger.logDeveloper(TAG, "Scheduling api call for retry")
-        if (pendingRetriesCount() < MAX_FAILED_CALLS || request.isSessionRequest()) {
-            val scheduleJob = retryQueue.isEmpty()
-            val cachedPayloadName = cacheManager.savePayload(payload)
-            val failedApiCall = DeliveryFailedApiCall(request, cachedPayloadName)
-            retryQueue.add(failedApiCall)
-            cacheManager.saveFailedApiCalls(retryQueue)
-
-            // By default there are no scheduled retry jobs pending. If the retry queue was initially empty, try to schedule a retry.
-            if (scheduleJob) {
-                scheduleFailedApiCallsRetry(RETRY_PERIOD)
+
+        val endpoint = Endpoint.fromEmbraceUrl(request.url)
+        endpoint?.let { e ->
+            if (isBelowRetryLimit(e)) {
```

Comment:
This works fine, ideally I'd like to change the strategy so the oldest retry gets dropped, rather than everything new. Also, we should ideally record somewhere that a retry was dropped. Not something to do in this PR

Reply:
Why are new attempts more valuable than the oldest ones?

Reply:
Good question - I don't think it's a clear cut distinction. I'd say they're slightly more valuable as folks care about things that happened recently. Also, there's less chance we're dealing with a bad scenario, like data corruption meaning an old session was stuck on a device & is unsendable.

Reply:
@eliabsisay I created [this Story](https://www.notion.so/embraceio/DCI-1678d2876bc247a8951b7ee79f25afe2?p=c24c5a8772ec4089890531bd7a693d72&pm=s) to work on it.